core/src/js/utilities/events.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * Create an event with the specified name in a browser-agnostic way.
+ * @param {string} eventName - the name of the event
+ * @return {Event} - instance of event which can be dispatched / listened for
+ */
+export const createEvent = (eventName = '') => {
+  if (typeof eventName !== 'string' || eventName.length <= 0) {
+    return null;
+  }
+  if (typeof Event == 'function') { // modern browsers
```

Comment:
```suggestion
  // Modern browsers.
  if (typeof Event == 'function') {
```

Reply:
I think there's room for both styles of commenting. I like comments on their own line when they're introducing a distinct chunk of code. That style of commenting helps me chunk up code mentally, in part because it interrupts my reading of the code. I like comments at the end of the line when they're just asides to help clarify a particular line or two of code. They're off to the side, so they don't interrupt my reading of the code.

Reply:
Checking the top packages on npmjs.org

lodash: Seems to be above but mostly just has function comments
request: mixed, some above and some inline
chalk: mixed, some above and some inline
react: above
debug: above
moment: mixed, some above and some inline
express: mixed, commonly inline.

Checking wordpress standards it appears they use a modified jQuery convention:
- https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/
- https://contribute.jquery.org/style-guide/js/#comments

> Comments are always preceded by a blank line. Comments start with a capital first letter, but don't require a period at the end, unless you're writing full sentences. There must be a single space between the comment token and the comment text.

I try to stick to the conventions of the project I intend to be working in as it keeps everything similar but it looks like it is fairly mixed in some of the top npm packages.

Reply:
> I try to stick to the conventions of the project I intend to be working in as it keeps everything similar but it looks like it is fairly mixed in some of the top npm packages.

I do try to stick to the conventions of the project I’m working in. But since this is the very first Javascript in the project, I thought there might be some flexibility around what those conventions are.

Also, to my mind, conventions are not ends in themselves. They’re there to make reading and understanding the code easier. I honestly cannot imagine a scenario where a comment beginning with a lower case letter vs. an upper case letter would affect anyone’s ability to read and understand the code. I therefore don’t think it’s in anyone’s best interests to bother having a convention about that. I accept that some people do make conventions about that, but to my mind that’s not a productive use of anyone’s time.

Of course discussing comment style at this length is also not a productive use of our time. I’ve changed the code per your requests.

Reply:
I'd argue that the very first of something is the best time to start a convention. Instead of letting something snowball into work later, nip it early and carry on. Starting early is also a benefit to the early contributors as they get the biggest say into what the convention should be.

I agree that capitalization and punctuation of comments is not a make or break scenario for readability but rather, my concerns stem from having consistency in style. Consistency plays a big part into readability and is good for the overall health of the project. Many other people much smarter than myself have had long discussions on the merits of each style and have created unified standards for their projects. I'd like to stay in line with the projects we intend to use this code in and when there isn't one, pick from other popular projects. 

My goal of these reviews is not to have long pedantic conversations about style, but rather, to call out and pick a standard when I see a difference. Please bear with this process as it gets quicker as we go.

Reply:
I trust that we both have the best interests of the project at heart. But I think the comment thread of what was meant to be a minor PR is likely not the best forum for proposing, discussing or determining these kinds of coding standards, especially when there's clearly some disagreement. If we want to invest time in coming to a shared consensus on standards, I'm happy to discuss it in a call, either 1-1 or in a regular meeting. Alternatively, I did propose including a coding standards issue in our upcoming Decanterpalooza, for just this reason. But in the meantime, for the sake of this PR, I believe I have made all the changes you requested. I'm happy to shake hands, reaffirm our shared commitment to writing excellent code that benefits Stanford, and go forth and conquer more components! :)

Reply:
That's a really good point. We could speed things up by spending some time hashing out the lint files. +1 to getting to those tickets sooner than later.